engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -196,15 +194,18 @@ public AnnotationProcessingOptions getAnnotationProcessingOptions() {
 			return Collections.emptySet();
 		}
 
-		Set<MetaConstraint<?>> classLevelConstraints = newHashSet();
+		List<ConstraintDescriptorImpl<?>> classLevelConstraintDescriptors = findConstraints( null, clazz.getDeclaredAnnotations(),
+				ConstraintLocationKind.TYPE );
 
-		// HV-262
-		List<ConstraintDescriptorImpl<?>> classMetaData = findClassLevelConstraints( clazz );
+		if ( classLevelConstraintDescriptors.isEmpty() ) {
+			return Collections.emptySet();
+		}
 
+		Set<MetaConstraint<?>> classLevelConstraints = newHashSet( classLevelConstraintDescriptors.size() );
```

Comment:
Nothing big, but maybe we could slightly modify the existing `convertToMetaConstraints()` for field to receive list of descriptors and a `ConstraintLocation` instead of a field. This way we could use it for class, field and parameter constraints. And the code would look even more consistent for these cases. WDYT?

Reply:
So, one of the thing is that I want to avoid creating locations if we don't need it. Having to pass a location would defeat that or I would have to move the empty test above in the call stack. I don't think it will help.

We could probably do things a bit differently but I think making things consistent will be easier once we really use `JavaBean` to abstract the `getDeclaringClass()` thing. We will have a `Constrainable` to pass and everything will be quite consistent (but we would still need several `convertToMetaConstraints()` methods).